engine/events/field_moves.asm

```diff
@@ -53,11 +64,19 @@ ShakeHeadbuttTree:
 	call ApplyTilemapInVBlank
 	xor a
 	ldh [hBGMapMode], a
-	call ClearSpriteAnims
-	ld hl, wShadowOAM + 36 * 4
-	ld bc, wShadowOAMEnd - (wShadowOAM + 36 * 4)
+	ld a, [wCurSpriteOAMAddr]
+	sub 4 * SPRITEOAMSTRUCT_LENGTH
+	cpl
+	add LOW(wShadowOAMEnd) + 1
+	ld c, a
+	ld b, 0
+	ld h, HIGH(wShadowOAM)
+	ld a, [wCurSpriteOAMAddr]
+	sub 12 * SPRITEOAMSTRUCT_LENGTH
```

Comment:
Lots of 12s and 13s in this file. Is that because there's a maximum of 12 (or 13?) weather particles? If so, that could be a constant on top of this file (or in weather_constants if it occurs in other files too).